rag_app/ingest.py

```diff
@@ -31,21 +41,30 @@ def batch_chunks(chunks, batch_size=20):
 def chunk_text(
     documents: Iterable[Document], window_size: int = 1024, overlap: int = 0
 ):
-    id = 0
     for doc in documents:
         for chunk_num, start_pos in enumerate(
-            range(0, len(doc.text), window_size - overlap)
+            range(0, len(doc.content), window_size - overlap)
         ):
             # TODO: Fix up this and use a Lance Model instead - have reached out to the team to ask for some help
             yield {
-                "id": id,
                 "doc_id": doc.id,
-                "chunk_num": chunk_num,
-                "start_pos": start_pos,
-                "end_pos": start_pos + window_size,
-                "text": doc.text[start_pos : start_pos + window_size],
+                "chunk_id": chunk_num,
+                "text": doc.content[start_pos : start_pos + window_size],
+                "post_title": doc.metadata["title"],
+                "publish_date": datetime.strptime(doc.metadata["date"], "%Y-%m"),
+                "source": doc.metadata["url"],
             }
-            id += 1
+
+
+def populate_chunks_with_counts(chunks: Iterable[TextChunk]):
+    counts = defaultdict(list)
+
+    for chunk in chunks:
+        counts[chunk["doc_id"]].append(chunk)
+
+    for doc in counts.keys():
+        for chunk in counts[doc]:
+            yield {**chunk, "post_chunk_count": len(counts[doc])}
```

Comment:
Not sure if there's a better way for this but I figured this would allow us to keep to the use of a generator

Reply:
ah id maybe ignore this cause you're right we're using a generator, just make the index without the post chunk count. 

lance db can write a query like 

SELECT doc_id, max(chunk_index)
FROM docs

Reply:
Hmm a brute force way might be to extract all of the unique doc_ids. do another filter query for all entries that match that doc_id and then work off that. 

Alternatively, we could have another table which stores documents. Everytime we update/store a new document, we run a query on the table of chunks to count the number of chunks that have that doc.id

Reply:
I managed to fix up the lancedb query but that added a pandas and a duckdb dependency.